### DISCUSSION TITLE
OSD won't launch

### DISCUSSION THREAD

**Discussion by kellyschrock** [svpcom/wfb-ng#296] [Questions]
Hi, I have the latest version of WFB-ng and it's working well, except I can't seem to get the OSD to work. I have it set to start on boot with `sudo systemctl enable osd` and it attempts to launch, but fails. Under `/home/pi/wfb_ng_osd`, I find an `osd` executable I can run but when I launch it with `./osd`, I get this:

```
Usage: mavlink_port=14551, rtp_port=5600, codec=h264, rtp_jitter=5, use_vaapi=0, use_xv=0, screen_width=1920
* failed to add service - already in use?
```
Any idea what I might be doing wrong? Thanks...

**Comment by svpcom** [comment]
What HW do you have? RPI3 or RPI4?

**Reply by kellyschrock** [reply]
Sorry, I should have specified that. Both sides (air, ground) are RPI4.

**Reply by svpcom** [reply]
Native OSD only supported on RPI3. Broadcom removed OpenVG support on RPI4, so OSD is only available in gstreamer pipeline via cairooverlay under X11 session.

**Reply by kellyschrock** [reply]
Ah, well that's unfortunate! 

I actually modded WFB to get it to expose an SSID I could attach to with GCS apps on the Pi's internal wifi card on 2.4Ghz, so I can get video with telemetry data on a different screen (any number of apps can connect and see video/telem). But the video is a little choppy (presumably from the handoff between wlan0 and wlan1). 

Overall, I'm more than pleased with WFB-ng!

**Reply by lida2003** [reply]
@kellyschrock Did you have tested lantency on RPI4?

**Reply by kellyschrock** [reply]
@lida2003 Not formally, but it's very good in my experience. I would estimate it to be around maybe 150ms, maybe less.

**Reply by lida2003** [reply]
> @lida2003 Not formally, but it's very good in my experience. I would estimate it to be around maybe 150ms, maybe less.

what type of plane are you flying, copter(quard) or fixed wind ? 

I'm NOT sure if ~100ms is OK for intensive maneuver? 

PS： Maybe I have to assemble a quard to test myself :)

**Reply by kellyschrock** [reply]
What I'm flying with this right now is a copter I designed from scratch and built. It's not an FPV copter, it has 10" props, 800KV motors, runs ArduPilot, and has a Pi on it that I use as a companion computer. I mainly fly missions with it. It's not really suitable for intensive maneuvers. :-) The latency for the type of flying I do with this copter is fine, but I wouldn't be comfortable flying FPV with that kind of latency.

I did some modifications to WFB for the copter side to make it work with my other companion-computer software. One thing on my todo list is to try and increase the mavlink bandwidth, since mission uploads (which utilize a UDP connection from a tablet or PC to the ground station to the vehicle) are a bit slow. Other than that (and not being able to see the OSD on the Pi's screen), it's great!
